packages/fbjs/src/functional/partitionObjectByKey.js

```diff
@@ -18,10 +18,10 @@ var partitionObject = require('partitionObject');
  * `whitelistObjectKeys`, but eventually keeping all the keys. Returns a tuple
  * of objects `[first, second]`.
  */
-function partitionObjectByKey(
-  source: Object,
+function partitionObjectByKey<Tv>(
+  source: {[key: string]: Tv},
   whitelist: Set<string>
-): Array<Object> {
+): [{[key: string]: Tv}, {[key: string]: Tv}] {
```

Comment:
It seems like the caller of this function might need to do a type cast. My guess is that Flow has to define `Tv` as the most generic supertype of all the object values, which is often something like `mixed` for React component props.

So the caller either should type cast the argument as `Object` or `{[key: string]: any}`, or type cast the returned partitions as `{[key: string]: whateverMakesSense}`.

Flow might be able to do something smart in this specific instance with $Keys, not sure if the complexity is worth it compared to using `Object` though.